### PR TITLE
net: lib: nrf_cloud: remove P-GPS prediction periods which do not work

### PR DESCRIPTION
**Pull request by tokangas**
This PR proposes to remove the P-GPS Kconfig option `CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD`. I haven't paid much attention to this before, but now that I started looking into it, I'm not sure what the purpose of this configuration is.

The range for this option is 60 - 480 minutes and the default is 240 minutes. 4 hours makes sense, because each ephemeris/prediction is valid for 4 hours.

I tried different values and 60 minutes and 480 minutes don't look too good. When the option is set to 60 minutes, the cloud still sends predictions with 240 min period and download is aborted by the P-GPS library because of this mismatch. When the option is set to 480 minutes, at least the first received ephes are exactly the same as with 240 minutes, so the ephes are not valid for a longer time. Because of this, the ephemerides just expire at the end of the 240 min period and no new ephes are injected by the library for another four hours. I'm also not sure whether our GNSS implementation would handle ephes with a longer fit interval properly anyway. The ephes received from GPS satellite broadcast have always 4 hour fit interval.

120 minutes could make some sense and it looks like it works properly. Using only the middle two hours of each ephemeris would in theory improve accuracy. However, the ephemerides are anyway predictions, so most likely any theoretical improvement in accuracy is lost in accumulated errors due to predicting the satellite location in the future.

@plskeggs, do you know a good reason why this configuration is needed? To me it looks like we should just always stick to the 4 hour period.

### CODE REVIEW

File: doc/nrf/libraries/networking/nrf_cloud_pgps.rst [hunk 1]
```diff
@@ -137,7 +136,7 @@ If the :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_REQUEST_UPON_INIT` option is disab
 In these cases, predictions might be unavailable until a connection is established to the cloud.
 
 .. note::
-   Each prediction requires 2 KB of flash. For prediction periods of 240 minutes (four hours), and with 42 predictions per week, the flash requirement adds up to 84 KB.
+   Each prediction requires 2 kB of flash. With 42 predictions per week, the flash requirement adds up to 84 kB.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_pgps.rst:139]:
```suggestion
   Each prediction requires 2 kB of flash.
   With 42 predictions in a week, the flash requirement adds up to 84 kB.
```

File: subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_pgps [hunk 2]
```diff
@@ -25,25 +25,14 @@ config NRF_CLOUD_PGPS_REQUEST_UPON_INIT
...
 	int "Desired number of predictions"
 	range 2 84
 	default 42
 	help
-	  This value, together with the prediction period, determines the
-	  overall timespan that the GPS predictions will cover. The
-	  default values allows for one week of predictions. Odd numbers
-	  are not allowed.
+	  This value determines the overall timespan that the GPS predictions
+	  will cover. The default values allows for one week of predictions.
```
Comment (peknis) [inline subsys/net/lib/nrf_cloud/Kconfig.nrf_cloud_pgps:34]:
```suggestion
	  will cover. The default values allow for one week of predictions.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Comment by plskeggs** [comment]
@tokangas -- I don't recall now why I set the range as low as 60 or as high as 480, but the idea was to cover at a minimum 120 and 240 as choices, since HERE provides predictions to the nRF Cloud backend in 120 minute intervals.  The default 240 is actually skipping every other prediction they give us.

At the time I implemented this, it wasn't clear whether allowing 120 minute intervals (at the expense of 2X the flash requirements) would benefit some use cases or not.

I suggest before removing this feature, your team could consider testing accuracy of 120 vs. 240 a week or 2 weeks and see if 120 provides any benefit at all.

**Comment by tokangas** [comment]
> @tokangas -- I don't recall now why I set the range as low as 60 or as high as 480, but the idea was to cover at a minimum 120 and 240 as choices, since HERE provides predictions to the nRF Cloud backend in 120 minute intervals. The default 240 is actually skipping every other prediction they give us.

Ok, that's how I though it works as well.

> I suggest before removing this feature, your team could consider testing accuracy of 120 vs. 240 a week or 2 weeks and see if 120 provides any benefit at all.
> 
> At a minimum, changing this to be fixed values of 120 vs. 240 would be ok with me.

Yes, at least the configuration should be changed so that only 120 and 240 min periods can be selected. I'll see if we could do a test run with two devices, one using 120 min and one using 240 min period.

**Review by hiltunent — APPROVED** [review]

**Comment by tokangas** [comment]
A one week test with both 120 min and 240 min prediction period shows marginal improvement in accuracy and time to fix with 120 min period. Not all CEPs were better, but because 120 min period may improve the performance slightly, maybe we should keep the option.

I'll change the implementation so, that it's possible to choose between 120 min and 240 min prediction periods.

**Comment by tokangas** [comment]
@plskeggs @peknis @hiltunent Reworked the PR so, that instead of removing the prediction period configuration it's now possible to choose between the two values which are usable, 240 minutes and 120 minutes. Please review.

**Review by peknis — APPROVED** [review]

**Review by hiltunent — APPROVED** [review]

File: doc/nrf/libraries/networking/nrf_cloud_pgps.rst [hunk 3]
```diff
@@ -63,7 +63,7 @@ Configure one of the following options to control the network transport for down
 
 Configure these additional options to refine the behavior of P-GPS:
 
-* :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD`
+* :kconfig:option:`CONFIG_NRF_CLOUD_PGPS_PREDICTION_PERIOD_120_MIN`
```
[the remark this answers is not in the document]
Comment (tokangas) [reply doc/nrf/libraries/networking/nrf_cloud_pgps.rst]: Thanks, good point. It's updated now.

**Review by trantanen — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]
LGTM.  Thanks for doing this.

**Pull request merged**